maven-compat/src/main/java/org/apache/maven/repository/legacy/DefaultUpdateCheckManager.java

```diff
@@ -350,60 +350,29 @@ private Properties read( File touchfile )
 
         synchronized ( touchfile.getAbsolutePath().intern() )
         {
-            FileInputStream in = null;
-            FileLock lock = null;
-
             try
             {
                 Properties props = new Properties();
 
-                in = new FileInputStream( touchfile );
-                lock = in.getChannel().lock( 0, Long.MAX_VALUE, true );
-
-                getLogger().debug( "Reading resolution-state from: " + touchfile );
-                props.load( in );
-
-                lock.release();
-                lock = null;
+                try ( FileInputStream in = new FileInputStream( touchfile ) )
+                {
+                    try ( FileLock lock = in.getChannel().lock( 0, Long.MAX_VALUE, true ) )
+                    {
+                        getLogger().debug( "Reading resolution-state from: " + touchfile );
+                        props.load( in );
 
-                in.close();
-                in = null;
+                        lock.release();
+                        return props;
+                    }
+                }
 
-                return props;
             }
             catch ( IOException e )
             {
                 getLogger().debug( "Failed to read resolution tracking file " + touchfile, e );
 
                 return null;
             }
-            finally
-            {
-                if ( lock != null )
-                {
-                    try
-                    {
-                        lock.release();
-                    }
-                    catch ( IOException e )
-                    {
-                        getLogger().debug( "Error releasing shared lock for resolution tracking file: " + touchfile,
-                                           e );
-                    }
-                }
-
-                if ( in != null )
-                {
-                    try
-                    {
-                        in.close();
-                    }
-                    catch ( IOException e )
-                    {
-                        getLogger().debug( "Error closing FileChannel for resolution tracking file: " + touchfile, e );
-                    }
-                }
-            }
         }
     }
 
```

Comment:
I think we don't case about `maven-compat` anymore so I wouldn't really touch unless necessary. @gnodet @cstamas WDYT?